### PR TITLE
Youtube plugin 

### PR DESCRIPTION
**Pull request by smitfire**
Finally wrapped some initial spec. So i've kept this in a separate collapse panel. As shown below.
CleanShot 2021-07-15 at 11 01 52
The left part is the 'thumbnailUrl' property. Followed by Name, Description -> Duration, UploadDate.

Below is the resource payload. You can test by going to a project, adding a new resource with this or other data like this, selecting it from the resource list and clicking the video dropdown. 
`{
  "@context": {
    "VideoObject": "https://schema.org/VideoObject",
    "description": "http://schema.org/description",
    "duration": "http://schema.org/duration",
    "embedUrl": "http://schema.org/embedUrl",
    "name": "http://schema.org/name",
    "thumbnailUrl": "https://schema.org/thumbnailUrl",
    "uploadDate": "https://schema.org/uploadDate",
    "video": "https://schema.org/itemListElement"
  },
  "@type": [
    "Entity",
    "Dataset"
  ],
  "name": "All Fields",
  "video": [
    {
      "@type": "VideoObject",
      "description": "a description of the video",
      "duration": "PT1M54S",
      "embedUrl": "https://www.youtube.com/watch?v=ysz5S6PUM-U",
      "name": "cool brain video",
      "thumbnailUrl": [
        "https://fdn.gsmarena.com/imgroot/news/21/02/youtube-announces-planned-updates/-1220x526/gsmarena_000.jpg",
        "https://example.com/photos/4x3/photo.jpg",
        "https://www.xda-developers.com/files/2019/06/YouTube-Vanced-New-1024x672.png"
      ],
      "uploadDate": "2016-03-31T08:00:00+08:00"
    }
  ]
}`

A modal is displayed when the name of the video is clicked. As seen below:
CleanShot 2021-07-15 at 11 05 17

### CODE REVIEW

**Review by nicwells — DISMISSED** [review]

**Comment by nicwells** [comment]
Looking great! A couple of things:

1. Video plugin shows up regardless of the resource, should show up only if it has an embedded YouTube video.
2. Video continues playing if you close it, is it possible to stop on close of the modal? Perhaps it could auto-start on opening too?

**Comment by smitfire** [comment]
> Looking great! A couple of things:
> 
> 1. Video plugin shows up regardless of the resource, should show up only if it has an embedded YouTube video.
> 2. Video continues playing if you close it, is it possible to stop on close of the modal? Perhaps it could auto-start on opening too?

yeah no problem, thanks for reviewing.

File: src/shared/containers/VideoPluginContainer.tsx [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+const { Panel } = Collapse;
+
+type VideoProps = {
+  orgLabel: string;
+  projectLabel: string;
+  resource: Resource;
+};
+
+type VideoObject = {
+  name: String;
```
Comment (Stafeeva) [inline src/shared/containers/VideoPluginContainer.tsx:19]: is there a difference between `String` and `string`? can we use just `string` to be consistent? :)

File: src/shared/containers/VideoPluginContainer.tsx [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+  };
+  const loadVideo = async () => {
+    const videoResource = (await nexus.Resource.get(
+      orgLabel,
+      projectLabel,
+      encodeURIComponent(resource['@id'])
+    )) as Resource<{
+      video: VideoObject[];
+    }>;
+    const videoData = videoResource['video']
```
Comment (Stafeeva) [inline src/shared/containers/VideoPluginContainer.tsx:64]: Can we use dot notation everywhere? Like `videoResource.video`.
Dot notation is more performant and easier to read. There is no need to use bracket notation as there there are no special character in the property name.

File: src/shared/containers/VideoPluginContainer.tsx [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+                    >
+                      {item.name}
+                    </Button>
+                  }
+                  description={item.description}
+                />
+              </List.Item>
+            )}
+          />
+          {selectedVideo && !!selectedVideo['name'] ? (
```
Comment (Stafeeva) [inline src/shared/containers/VideoPluginContainer.tsx:113]:
```suggestion
          {selectedVideo && !!selectedVideo.name ? (
```

File: src/shared/containers/VideoPluginContainer.tsx [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+            <Modal
+              title={selectedVideo['name']}
```
Comment (Stafeeva) [inline src/shared/containers/VideoPluginContainer.tsx:115]:
```suggestion
              title={selectedVideo.name}
```

File: src/shared/containers/VideoPluginContainer.tsx [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+              bodyStyle={{ padding: 0 }}
+              visible={isModalVisible && !!selectedVideo}
+              onOk={handleOk}
+              onCancel={handleCancel}
+              width={640}
+              footer={null}
+            >
+              {!!selectedVideo['embedUrl'] ? (
```
Comment (Stafeeva) [inline src/shared/containers/VideoPluginContainer.tsx:123]: dot notation please :)

**Review by Stafeeva — APPROVED** [review]

**Review by nicwells — APPROVED** [review]
Tested with latest version with both single and multiple videos. Looks good

**Review by dhaneshnm — APPROVED** [review]

**Pull request merged**